### DISCUSSION TITLE
Border with color gradient

### DISCUSSION THREAD

**Discussion by abodi-mls** [chakra-ui/chakra-ui#5250] [Q&A]
Hello all,

I'm struggling to get my borders styled with a gradient color.

Basically, I want my component's border to be a linear gradient and the content background to be transparent (i.e. match the component that contains it).

Also, when hovering over the element, I want its content background to fill up with gradient color and match the border. But that I could easily get with `bgGradient`, obviously.

What is the best way to achieve this using Chakra?

Thank you in advance.

**Comment by primos63** [comment]
See the `gradient` variant I created in my [CodeSandbox](https://codesandbox.io/s/chakra-ui-theme-extension-w5u2n?file=/src/theme/Button/index.js).

**Reply by WenLonG12345** [reply]
Hi, your solution is awesome for <Button> components! but how about normal <Box>? How can I apply coloured border gradient for <Box> ?

**Comment by peterburgs** [comment]
really looking forward to a new answer from this thread... 😢

**Comment by charliemday** [comment]
Could you place a white box inside a padded component with a background gradient? Something along the lines of:

```
  <Box
    rounded="2xl"
    w="full"
    bgGradient="linear(to-r, green.200, pink.500)"
    p={1}
  >
    <Box bg="white" rounded="xl">
      <Text>This is some text</Text>
    </Box>
  </Box>
```

**Comment by Neschadin** [comment]
I realize this is an old topic, but maybe someone will find this useful in the future:

```
<Center
      borderRadius='8px'
      border='2px'
      borderColor='transparent'
      bg='linear-gradient(#141414 0%, #2e2e2e 50%, #141414 100%) padding-box, 
          linear-gradient(#272726 0%, #4d4d4c 50%, #161616 100%) border-box'
    >
    </Center>
```

just set your values and enjoy  o:)
